pages/reserve-overview.page.tsx

```diff
@@ -22,21 +22,28 @@ export default function ReserveOverview() {
   const underlyingAsset = router.query.underlyingAsset as string;
   const { breakpoints } = useTheme();
   const lg = useMediaQuery(breakpoints.up('lg'));
+  const [reserve, setReserve] = useState<ComputedReserveData>();
 
   const [mode, setMode] = useState<'overview' | 'actions' | ''>('');
 
   useEffect(() => {
     if (!mode) setMode('overview');
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [lg]);
 
-  const reserve = reserves.find(
-    (reserve) => reserve.underlyingAsset === underlyingAsset
-  ) as ComputedReserveData;
+    if (router.isReady && reserves.length) {
+      const reserveFound = reserves.find(
+        (reserve) => reserve.underlyingAsset === underlyingAsset
+      ) as ComputedReserveData;
+
+      if (reserveFound) setReserve(reserveFound);
+      else router.replace('/404');
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [lg, router.isReady, reserves]);
```

Comment:
@grothem @JoaquinBattilana Do you think it makes sense to change the `reserves` dependency here to `reserves.length` so this useEffect doesn't re-render on every `useCurrentTimestamp` update?

Reply:
Sorry, just getting around to take a look at this. For the dependency, could we just use the underlying asset?

I think the overall approach makes sense, just a couple of suggestions:

```ts
if (router.isReady && reserves.length) {
...
}
```
I think this could be simplified, and only check if the reserves are loading. The `useAppDataContext` hook has a `loading` property that you could use here. I think the router will always be ready before the reserve data is loaded so it's redundant.

```ts
const reserveFound = reserves.find(
   (reserve) => reserve.underlyingAsset === underlyingAsset
) as ComputedReserveData;
```
I know this was existing, but it'd be awesome to get this cleaned up. Now that we know the reserves are loaded, we can get rid of the `as ComputedReserveData` type assertion.

The other thing I noticed that this change causes is that we'll lose some of our loading states in some of the child components because we were always rendering before, it's just that sometimes the reserve was undefined. I think we probably need to refactor some of that loading logic, but that'd probably be a different task.

Thanks for taking a look at this @Jdecristi !